test/e2e/e2e_test.go

```diff
@@ -1633,6 +1633,191 @@ var _ = ginkgo.Describe("e2e ingress gateway traffic validation", func() {
 	})
 })
 
+// Validate pods can reach a network running in a container's ipv6 loopback address via
+// an external gateway running on eth0 of the container without any tunnel encap.
+// Next, the test updates the namespace annotation to point to a second container
+// (emulating a second gateway) and validates ipv6 external gateway functionality
+var _ = ginkgo.Describe("e2e non-vxlan ipv6 external gateway and update validation", func() {
+	const (
+		svcname             string = "multiple-novxlan-ipv6-externalgw"
+		exGWRemoteIpAlt1    string = "fd01:9:244:3::5"
+		exGWRemoteIpAlt2    string = "fd02:8:244:4::5"
+		ovnContainer        string = "ovnkube-node"
+		gwContainerNameAlt1 string = "gw-novxlan-ipv6-test-container-alt1"
+		gwContainerNameAlt2 string = "gw-novxlan-ipv6-test-container-alt2"
+	)
+
+	f := framework.NewDefaultFramework(svcname)
+
+	type nodeInfo struct {
+		name   string
+	}
+
+	var (
+		workerNodeInfo nodeInfo
+	)
+
+	ginkgo.BeforeEach(func() {
+
+		// retrieve worker node names
+		nodes, err := e2enode.GetBoundedReadySchedulableNodes(f.ClientSet, 3)
+		framework.ExpectNoError(err)
+		if len(nodes.Items) < 3 {
+			framework.Failf(
+				"Test requires >= 3 Ready nodes, but there are only %v nodes",
+				len(nodes.Items))
+		}
+
+		workerNodeInfo = nodeInfo{
+			name:   nodes.Items[1].Name,
+		}
+	})
+
+	ginkgo.AfterEach(func() {
+		// tear down the containers simulating the gateways
+		if cid, _ := runCommand("docker", "ps", "-qaf", fmt.Sprintf("name=%s", gwContainerNameAlt1)); cid != "" {
+			if _, err := runCommand("docker", "rm", "-f", gwContainerNameAlt1); err != nil {
+				framework.Logf("failed to delete the gateway test container %s %v", gwContainerNameAlt1, err)
+			}
+		}
+		if cid, _ := runCommand("docker", "ps", "-qaf", fmt.Sprintf("name=%s", gwContainerNameAlt2)); cid != "" {
+			if _, err := runCommand("docker", "rm", "-f", gwContainerNameAlt2); err != nil {
+				framework.Logf("failed to delete the gateway test container %s %v", gwContainerNameAlt2, err)
+			}
+		}
+	})
+
+	ginkgo.It("Should validate IPv6 connectivity (without vxlan) before and after updating the namespace annotation to a new external gateway", func() {
+
+		var (
+			pingSrc string
+			validIP net.IP
+			exGWRemoteCidrAlt1 = fmt.Sprintf("%s/24", exGWRemoteIpAlt1)
+			exGWRemoteCidrAlt2 = fmt.Sprintf("%s/24", exGWRemoteIpAlt2)
```

Comment:
better /64 for ipv6, right?

Reply:
thanks! good catch I forgot to swap out the v4 mask.